src/interpreter/interpreter.rs

```diff
@@ -28,24 +444,99 @@ pub fn execute(stmt: &Statement, env: Environment) -> Result<Environment, ErrorM
         Statement::Assignment(name, exp) => {
             let value = eval(exp, &env)?;
             let mut new_env = env;
-            new_env.insert(*name.clone(), value);
-            Ok(new_env.clone())
+            match value {
+                EvalResult::CInt(val) => {
+                    new_env.insert(*name.clone(), EnvValue::CInt(val));
+                }
+                EvalResult::CReal(val) => {
+                    new_env.insert(*name.clone(), EnvValue::CReal(val));
+                }
+                EvalResult::Bool(val) => {
+                    new_env.insert(*name.clone(), EnvValue::Bool(val));
+                }
+                EvalResult::List(val) => {
+                    new_env.insert(*name.clone(), EnvValue::List(val));
+                }
+                EvalResult::None => {
+                    new_env.insert(*name.clone(), EnvValue::None);
+                }
+            }
+            Ok(new_env)
```

Comment:
doesn't something like the following (I have not compiled/run, so there may be mistakes in my reasoning) look simpler/more concise?

```rust
Statement::Assignment(name, exp) => {
        let value = eval(exp, &env)?;
        let mut new_env = env;

        // Directly insert the value after cloning the name
        new_env.insert(name.clone(), match value {
            EvalResult::CInt(val) => EnvValue::CInt(val),
            EvalResult::CReal(val) => EnvValue::CReal(val),
            EvalResult::Bool(val) => EnvValue::Bool(val),
            EvalResult::List(val) => EnvValue::List(val),
            EvalResult::None => EnvValue::None,
        });

        Ok(new_env)
    }
    ```